pyphi/actual.py

```diff
@@ -410,14 +410,22 @@ def find_causal_link(self, direction, mechanism, purviews=False,
         if not purviews:
             max_ria = _null_ac_ria(self.mechanism_state(direction),
                                    direction, mechanism, None)
+            return CausalLink(max_ria)
         else:
-            # This max should be most positive
-            max_ria = max(self.find_mip(direction, mechanism, purview,
-                                        allow_neg)
-                          for purview in purviews)
-
-        # Construct the corresponding CausalLink
-        return CausalLink(max_ria)
+            # Finds rias with maximum alpha
+            all_ria = [self.find_mip(direction, mechanism, purview, allow_neg)
+                       for purview in purviews]
+            max_alpha = max(all_ria).alpha
+            max_rias = [ria for ria in all_ria if ria.alpha == max_alpha]
+
+            # Selected rias whose purview is not a superset of any other
+            def is_not_superset(purview):
+                return np.all([(not set(purview).issuperset(set(purview2))) or
+                               (set(purview) == set(purview2)) for purview2 in purviews])
+                               
+            purviews = [ria.purview for ria in max_rias]
+            extended_purview = filter(is_not_superset, purviews)
+            return CausalLink(max(max_rias), tuple(extended_purview))
```

Comment:
Call keyword arguments by name.

Also, you can simplify this a bit by assigning `max(ria)` to a variable, so you don't have to call `max()` twice:
```python
# Finds rias with maximum alpha
all_rias = [self.find_mip(direction, mechanism, purview, allow_neg=allow_neg)
            for purview in purviews]
max_ria = max(all_rias)
max_rias = [ria for ria in all_ria if ria.alpha == max_ria.alpha]

# ...

return CausalLink(max_ria, extended_purview=filter(is_minimal, purviews))
```